users/tests/test_unit_users_function_view.py

```diff
@@ -0,0 +1,73 @@
+from rest_framework.reverse import reverse
+
+from django.test import TestCase
+
+from users.common.strings import CustomValidationErrorText
+from users.common.utils import create_user_using_full_clean_and_save
+
+
+class ChangePasswordTests(TestCase):
+    def setUp(self):
+        self.user_password = 'userpasswd'
+        self.user = create_user_using_full_clean_and_save('testuser@codepoets.it', '', '', '', self.user_password)
+
+    def test_change_user_password_view_should_change_user_password_on_post(self):
+        data = {
+            'old_password': self.user_password,
+            'new_password1': 'newuserpasswd',
+            'new_password2': 'newuserpasswd',
+        }
+        self.client.login(
+            email=self.user.email,
+            password=self.user_password,
+        )
+        response = self.client.post(
+            reverse('change_password'),
+            data,
+        )
+        self.user.refresh_from_db()
+        self.assertEqual(response.status_code, 302)
+        self.assertTrue(self.user.check_password(data['new_password1']))
+
+    def test_change_user_password_view_should_not_change_user_password_when_old_password_is_incorrect(self):
+        data = {
+            'old_password': 'wronguserpasswd',
+            'new_password1': 'newuserpasswd',
+            'new_password2': 'newuserpasswd',
+        }
+        self.client.login(
+            email=self.user.email,
+            password=self.user_password,
+        )
+        response = self.client.post(
+            reverse('change_password'),
+            data,
+        )
+        self.user.refresh_from_db()
```

Comment:
Is this necessary here?

Reply:
@kbeker No, it's not. Just to be unified

Reply:
I think it is needed here. You have to refresh the user to check if a refreshed user password did not change. Otherwise it might actually change, but the test will think it does not, because it has outdated user instance.